### PR TITLE
RA-331: Design Match Data Finder

### PR DESCRIPTION
**Pull request by aameijide**

### CODE REVIEW

File: VAS.UI.Gtk2/UI/Component/DatePicker.cs [hunk 1]
```diff
@@ -51,6 +51,18 @@ public DatePicker ()
 			}
 		}
 
+		/// <summary>
+		/// Sets the background color for a DatePicker in a particular state.
+		/// It also changes his childs background.
+		/// </summary>
+		/// <param name="state">Type.</param>
+		/// <param name="color">Color.</param>
+		public new void ModifyBg (Gtk.StateType state, Gdk.Color color)
```
Comment (aameijide) [inline VAS.UI.Gtk2/UI/Component/DatePicker.cs]: I'm overwriting the ModifyBg from Gtk.Bin, it's for that reason that Gdk.Color is used

File: VAS.UI.Gtk2/UI/TreeView/CellRendererButtonImage.cs [hunk 2]
```diff
@@ -16,30 +16,38 @@
...
-
+		const int BUTTON_WIDTH = 35;
+		const int BUTTON_HEIGHT = 35;
+		const int SPACING = 5;
+		const int LINE_WIDTH = 1;
+		static double offsetX, offsetY = 0;
+		static int cellWidth, cellHeight = 0;
+		static Point cursor;
 		public event ClickedHandler Clicked;
+		static bool downloadButtonPrelighted = false;
```
Comment (aameijide) [inline VAS.UI.Gtk2/UI/TreeView/CellRendererButtonImage.cs]: Copy and paste fail, it should be buttonPrelighted

File: VAS.UI.Gtk2/UI/TreeView/CellRendererButtonImage.cs [hunk 3]
```diff
@@ -61,32 +69,77 @@ public override void GetSize (Widget widget, ref Rectangle cell_area, out int x_
...
-				tk.StrokeColor = App.Current.Style.PaletteText;
-				tk.DrawImage (pos, width, height, Icon, ScaleMode.AspectFit);
+				tk.DrawRectangle (pos, BUTTON_WIDTH, BUTTON_HEIGHT);
+				tk.DrawImage (imagePos, BUTTON_WIDTH - (SPACING * 2), BUTTON_HEIGHT - (StyleConf.FilterTreeViewOnlyTopOffset * 2), Icon, ScaleMode.AspectFit);
 				tk.End ();
 				tk.Context = null;
 			}
 		}
+
+		bool IsCursorInsideArea (Point p, double width, double height)
```
Comment (aameijide) [inline VAS.UI.Gtk2/UI/TreeView/CellRendererButtonImage.cs]: Moved to Point

File: VAS.UI.Helpers.Gtk2/ButtonHelper.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			button.HeightRequest = App.Current.Style.ButtonDialogHeight;
+		}
+
+		static Image FillButton (Button button, Pixbuf icon, string toolTip, Command command, string buttonStyle)
+		{
+			button.Name = buttonStyle;
+			button.TooltipMarkup = toolTip;
+			Image image = button.Children.FirstOrDefault () as Image;
+			if (image == null) {
+				image = new Image ();
```
Comment (vguzmanp) [inline VAS.UI.Helpers.Gtk2/ButtonHelper.cs]: Check for null icon before creating the image.

**Pull request merged**